extensions/git/src/statusbar.ts

```diff
@@ -66,9 +68,23 @@ class SyncStatusBar {
 	constructor(private repository: Repository) {
 		repository.onDidRunGitStatus(this.onModelChange, this, this.disposables);
 		repository.onDidChangeOperations(this.onOperationsChange, this, this.disposables);
+
+		const onEnablementChange = filterEvent(workspace.onDidChangeConfiguration, e => e.affectsConfiguration('git.statusBarSync.enabled'));
+		onEnablementChange(this.updateEnablement, this, this.disposables);
+
 		this._onDidChange.fire();
 	}
 
+	private updateEnablement(): void {
+		const isEnabled = workspace.getConfiguration('git').get('statusBarSync.enabled');
+
+		if (isEnabled) {
+			this.state = { ... this.state, isEnabled: true };
+		} else {
+			this.state = { ... this.state, isEnabled: false };
+		}
```

Comment:
@joshleeb are you think about this suggestion? Without this one _if_ isn't simpler?
```ts
this.state = { ... this.state, isEnabled: isEnabled };
```